swift/ql/lib/codeql/swift/regex/Regex.qll

```diff
@@ -299,35 +299,21 @@ class NSStringRegexAdditionalFlowStep extends RegexAdditionalFlowStep {
  * Regex("(a|b).*").firstMatch(in: myString)
  * ```
  */
-abstract class RegexEval extends CallExpr {
-  /**
-   * Gets the input to this call that is the regular expression being evaluated.
-   * This may be a regular expression object or a string literal.
-   */
-  abstract DataFlow::Node getRegexInput();
-
-  /**
-   * Gets the input to this call that is the string the regular expression is evaluated on.
-   */
-  abstract DataFlow::Node getStringInput();
-
-  /**
-   * Gets a dataflow node for the options input that might contain parse mode
-   * flags (if any).
-   */
-  DataFlow::Node getAnOptionsInput() { none() }
+class RegexEval extends CallExpr instanceof PotentialRegexEval {
+  RegexEval() { this.(PotentialRegexEval).doesEvaluate() }
```

Comment:
All these `this.(PotentialRegexEval)` can be replaced with `super` (and since `super = this` in charpreds this `this.(PotentialRegexEval)` can actually be replaced by `this`).